### DISCUSSION TITLE
Add Event Listener to nuxt 3

### DISCUSSION THREAD

**Discussion by casualmatt** [nuxt/nuxt#17610] [Questions] [answered]
Hello to all,

I succeed to integrate my live chat widget from [chatwoot](https://github.com/chatwoot/chatwoot), but to implement other functionality i need to add a event listener to w8 the [SDK](https://www.chatwoot.com/docs/product/channels/live-chat/sdk/setup) to load. 
```js
window.addEventListener("chatwoot:ready", function () {
  // Use window.$chatwoot here
  // ...
});

```
Any suggestion on how I could do that?

I try to add it to `mounted `and to `beforeMount`, but it didn't work.

**Comment by florian-lefebvre** [comment, marked as answer]
Weird, it should work. Here is a composable I made for a project, you can see how it's done: https://github.com/florian-lefebvre/portfolio/blob/v4/nuxt/composables/useWindowEventListener.ts

**Reply by casualmatt** [reply]
ty for the suggestion, i will try to do a composable for chatwoot.
